### PR TITLE
feat: boards and search

### PR DESCRIPTION
**Pull request by Boris713**
Changes-
1. Installed packages for backend
2. made call to database to populate boards on main page
3. search feature
4. added random image from lorem picsum

Screenshot 2024-06-18 at 9 03 00 PM
Screenshot 2024-06-18 at 9 03 48 PM
Screenshot 2024-06-18 at 9 04 09 PM
Screenshot 2024-06-18 at 9 04 21 PM

### CODE REVIEW

File: client/src/KudosRow.jsx [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
 import "./KudosRow.css";
 import KudosCard from "./KudosCard";
-const KudosRow = () => {
+
+const KudosRow = ({ searchTerm }) => {
+  const [boardInfo, setBoardInfo] = useState([]);
+
+  const populateCards = async () => {
+    try {
+      const resp = await fetch("http://localhost:3000/kudos/board");
```
Comment (akhilmandadi) [inline client/src/KudosRow.jsx]: Set the URL in a .env or config file

File: client/src/KudosRow.jsx [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
+      const data = await resp.json();
+      setBoardInfo(data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  useEffect(() => {
+    populateCards();
+  }, [searchTerm]);
```
Comment (akhilmandadi) [inline client/src/KudosRow.jsx]: Why do we need this dependency of searchTerm here?

File: package-lock.json
Comment (akhilmandadi) [inline package-lock.json:1]: add this to .gitignore. We shouldn't push this

File: server/node_modules/.package-lock.json
Comment (akhilmandadi) [inline server/node_modules/.package-lock.json:1]: add this to .gitignore

File: server/package-lock.json
Comment (akhilmandadi) [inline server/package-lock.json:1]: add this to .gitignore

File: server/routes/kudosRoutes.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            id: true,
+            imageUrl: true,
+            title: true,
+            author: true,
+            category: true
+        }
+    });
+    
+    return res.status(200).json(boards);
+}catch (error) {
```
Comment (akhilmandadi) [inline server/routes/kudosRoutes.js]: use formatter to format the file

**Review by akhilmandadi — CHANGES_REQUESTED** [review]

File: client/src/KudosRow.jsx (hunk 1 above)
Comment (Boris713) [reply client/src/KudosRow.jsx]: I created env.local and added to gitignore is that okay?

File: client/src/KudosRow.jsx (hunk 2 above)
Comment (Boris713) [reply client/src/KudosRow.jsx]: I thought to update everytime searchTerm changes but I dont think we need

**Comment by akhilmandadi** [comment]
Create a .gitignore for the server as well.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# General ignores
+node_modules/
+*.log
+# Environment files
+.env.local
+client/package.json
+client/package-lock.json
+package.json
```
Comment (akhilmandadi) [inline .gitignore:8]: You should push the package.json and ignore the server package-lock.json as well

**Pull request merged**